### PR TITLE
Improve `ThingActions` docs for UI-based rules & REST API support

### PR DESCRIPTION
**Pull request by florian-h05**

### CODE REVIEW

**Comment by florian-h05** [comment]
@lolodomo Can you please have a look?

File: developers/bindings/index.md [hunk 1]
```diff
@@ -630,9 +634,42 @@ In this case we refer to translation, see [i18n](../utils/i18n.html) support, in
...
+If you return a single value, add the `@ActionOutput` annotation.
+If you return multiple values, you do so by returning a `Map<String, Object>` and adding the `@ActionOutputs` annotation.
+Please note that it of course is still possible to return a `Map<?, Object>`, e.g. a `Map<Instant, Object>` for a collection of values with timestamps, but this return value cannot be used by rule actions (in UI-based rules) or the REST API.
+
+As the return value needs to be serialised to be sent over the REST API, only a limited set of output types are supported by the REST API.
+If an action method has a return value that has an unsupported type, the action's return value cannot be used by rule actions (in UI-based rules) or the REST API.
+The return value can still be used in automation languages.
+The supported output types are (see [`AnnotationActionHandler`](https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/AnnotationActionHandler.java)):
+
+- all primitive types, e.g. `boolean`, `int`, `float`, `String`, and their boxed counterparts
```
Comment (lolodomo) [inline developers/bindings/index.md:665]: Please add missing BigDecimal

**Review by lolodomo — CHANGES_REQUESTED** [review]
All this looks very good. I just noticed a forgotten possible type for output.

**Comment by lolodomo** [comment]
@florian-h05 @holgerfriedrich : that would be great to merge before releasing version 4.3.

File: developers/bindings/index.md (hunk 1 above)
Comment (florian-h05) [reply developers/bindings/index.md:665]: Just added, thanks for the catch.

**Comment by holgerfriedrich** [comment]
@lolodomo sorry, can't help here, this is docs repo.... I can only merge in core repo.

**Comment by florian-h05** [comment]
Ping @stefan-hoehn

**Review by stefan-hoehn — APPROVED** [review]
LGTM

**Pull request merged**